Source/player.cpp

```diff
@@ -3033,6 +3033,12 @@ void StripTopGold(Player &player)
 	player.HoldItem = tmpItem;
 }
 
+int GetManaShieldDamageReduction(int pnum)
+{
+	constexpr int8_t Max = 7;
+	return 24 - std::min(Players[pnum]._pSplLvl[SPL_MANASHIELD], Max) * 3;
+}
+

```

Comment:
This should be a member function on the `Player` class, (or at least take a `const Player &` argument).
```suggestion
int Player::GetManaShieldDamageReduction()
{
	constexpr int8_t Max = 7;
	return 24 - std::min(_pSplLvl[SPL_MANASHIELD], Max) * 3;
}
```

It'd then be used in ApplyPlrDamage like `player.GetManaShieldDamageReduction()`

Reply:
This method didn't work, but I tried directly adding the function definition into the player struct